hlapipeline/alignimages.py

```diff
@@ -0,0 +1,127 @@
+                                 "uvis":{
+                                     "fwhmpsf":0.076}}} # TODO: nothing. pipeline value in agreement with instrument handbook value.
+#-----------------------------------------------------------------------------------------------------------------------
+def main(input_list):
+    """Main calling function.
+    
+    Parameters
+    ----------
+    input_list: list
+        List of one or more input image(s) and/or association file(s) to align.
+
+    Returns
+    -------
+    Nothing for now.
+
+    """
+
+    # 1: Interpret input data and optional parameters
+
+    # 2: Apply filter to input observations to insure that they meet minimum criteria for being able to be aligned
+    imgList = input_list
+    # 3: Build WCS for full set of input observations
+    refwcs = amutils.build_reference_wcs(imgList)
+
+    # 4: Retrieve list of astrometric sources from database
+
+    # 5: Extract catalog of observable sources from each input image
+    extracted_sources = generate_source_catalogs(imgList,refwcs)
+
+    # 6: Cross-match source catalog with astrometric reference source catalog
+
+    # 7: Perform fit between source catalog and reference catalog
+#-----------------------------------------------------------------------------------------------------------------------
+def generate_source_catalogs(imgList,refwcs,**pars):
+    """Generates a dictionary of source catalogs keyed by image name.
+
+    Parameters
+    ----------
+    imgList: list
+        List of one or more calibrated fits images that will be used for source detection.
+    refwcs: astropy WCS object
+        Composite WCS information generated by build_reference_wcs() for the full set of input images.
+
+    Returns
+    -------
+    sourceCatalogDict: dictionary
+        a dictionary (keyed by image name) of two element dictionaries which in tern contain 1) a dictionary of the
+        detector-specific processing parameters and 2) an astropy table of position and photometry information of all
+        detected sources
+    """
+    sourceCatalogDict = {}
+
+    for imgName in imgList:
+        print("Image name:           ", imgName)
+
+        sourceCatalogDict[imgName] = {}
+
+        # open image
+        imgHDU = fits.open(imgName)
+        imgPrimaryHeader = imgHDU[0].header
+        instrument = imgPrimaryHeader['INSTRUME'].lower()
+        detector = imgPrimaryHeader['DETECTOR'].lower()
+
+        # get instrument/detector-specific image alignment parameters
+        sourceCatalogDict[imgName]["params"] = detector_specific_params[instrument][detector]
+
+        # Identify sources in image, convert coords from chip x, y form to reference WCS sky RA, Dec form.
+        imgwcs = HSTWCS(imgHDU, 1)
+        fwhmpsf_pix = sourceCatalogDict[imgName]["params"]['fwhmpsf']/imgwcs.pscale
+        sourceCatalogDict[imgName]["catalog_table"] = amutils.generate_source_catalog(imgHDU,refwcs,threshold = 1000,fwhm = fwhmpsf_pix)
```

Comment:
This version still does not provide any way to control any of the input parameters for amutils.generate_source_catalog.  If they can be passed in through '**pars', then those need to be passed along here as well.  That function may also need to be revised to enable more control, something which should be reviewed as well.